ui/src/app/pages/analysis/interactive-notebook.tsx

```diff
@@ -544,5 +498,79 @@ export const InteractiveNotebook = fp.flow(
         </div>
       );
     }
+
+    private renderPreviewContents() {
+      const { html, error } = this.state;
+      if (error) {
+        if (error instanceof ComputeSecuritySuspendedError) {
+          return (
+            <NotebookFrameError errorMode={ErrorMode.FORBIDDEN}>
+              <SecuritySuspendedMessage error={error} />
+            </NotebookFrameError>
+          );
+        }
+        if (error instanceof RuntimeStatusError) {
+          return (
+            <NotebookFrameError errorMode={ErrorMode.ERROR}>
+              {RUNTIME_ERROR_STATUS_MESSAGE_SHORT}
+            </NotebookFrameError>
+          );
+        }
+        const status = error instanceof Response ? error.status : 500;
+        if (status === 412) {
+          return (
+            <NotebookFrameError errorMode={ErrorMode.INVALID}>
+              Notebook is too large to display in preview mode, please use edit
+              mode or playground mode to view this notebook.
+            </NotebookFrameError>
+          );
+        } else {
+          return (
+            <NotebookFrameError errorMode={ErrorMode.ERROR}>
+              Failed to render preview due to an unknown error, please try
+              reloading or opening the notebook in edit or playground mode.
+            </NotebookFrameError>
+          );
+        }
+      }
+      if (html) {
+        return (
+          <iframe
+            id='notebook-frame'
+            style={styles.previewFrame}
+            srcDoc={html}
+          />
+        );
+      }
+      return <SpinnerOverlay />;
+    }
+
+    private startEditMode() {
+      const { nbName } = this.props.match.params;
+      const { appType } = getAppInfoFromFileName(nbName);
+      if (this.canStartRuntimes) {
+        if (!this.notebookInUse) {
+          if (appType === UIAppType.RSTUDIO) {
+            const { userApps } = this.props.userAppsStore;
+            const userApp = findApp(userApps, UIAppType.RSTUDIO);
+            if (userApp && userApp.status === AppStatus.RUNNING) {
+              window
+                .open(userApp.proxyUrls['rstudio-service'], '_blank')
```

Comment:
before opening RStudio, we need to call localize endpoint to start welder. 
https://github.com/all-of-us/workbench/blob/6a9cccdbd00c7473e265b2acc0bf690af5a8d0ed/ui/src/app/components/apps-panel/expanded-app.tsx#L175L180

Reply:
Ooo, thanks for catching that!

Reply:
It may be worth creating a small helper method since this code is very particular.